Hide orphaned tasks from follower stats

ywelsch opened the pull request:
CCR follower stats can return information for persistent tasks that are in the process of being cleaned up. This is problematic for tests where CCR follower indices have been deleted, but their persistent follower task is only cleaned up asynchronously afterwards. If one of the following tests then accesses the follower stats, it might still get the stats for that follower task.

In addition, some tests were not cleaning up their auto-follow patterns, leaving orphaned patterns behind. Other tests cleaned up their auto-follow patterns. As always the same name was used, it just depended on the test execution order whether this led to a failure or not. This commit fixes the offensive tests, and will also automatically remove auto-follow-patterns at the end of tests, like we do for many other features.

Closes #48700

ywelsch committed:
Hide orphaned tasks from follower stats

elasticmachine commented:
Pinging @elastic/es-distributed (:Distributed/CCR)

ywelsch committed:
ttests

tlrx reviewed (approved):
LGTM, thanks for fixing this

ywelsch commented:
PR testing shows that this is unfortunately still failing the test that it set out to fix. While I was able to reproduce with the steps outlined in https://github.com/elastic/elasticsearch/issues/48700 and fix that occurrence with this, there seems to be more to this.

ywelsch committed:
finally fix it

ywelsch commented:
Finally figured it out. Some tests were not cleaning up their auto-follow patterns, leaving orphaned patterns behind. Other tests cleaned up their auto-follow patterns. As always the same name was used, it just depended on the test execution order whether this led to a failure or not. I've fixed the offensive tests, and also added the ability to automatically remove auto-follow-patterns at the end of test, like we have for many other things.

dnhatn reviewed (approved):
LGTM. Thank you for chasing this test failure.
